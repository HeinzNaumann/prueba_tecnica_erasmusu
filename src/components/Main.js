import React, { useEffect, useState } from 'react'
import Loading from '../common/Loading'
import usePropertyType from '../hooks/useCategorias'
import styles from './Main.module.css'
const Main = () => {
	const { properties, loading, getPropertyType } = usePropertyType()
	useEffect(() => {
		getPropertyType('all')
	}, [])
	return (
		<>
			{loading ? (
				<Loading />
			) : (
				properties.map(property => (
					<div className={styles.propertyBox} key={property.id}>
						<img src={property.photoUrls.homecard}></img>
						<p>{property.title}</p>
						<div>
							<p>{property.pricePerMonth}</p>
							<div>
								<button></button>
								<button></button>
							</div>
						</div>
					</div>
				))
			)}
		</>
	)
}

export default Main
