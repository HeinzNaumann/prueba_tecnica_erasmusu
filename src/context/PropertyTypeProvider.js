import { useState, useEffect, createContext, useCallback } from 'react'
import axios from 'axios'
const PropertyTypeContext = createContext()

const PropertyTypeProvider = ({ children }) => {
	//const [propertyIds, setPropertyIds] = useState([])
	const [propertyIdsFilter, setPropertyIdsFilter] = useState([])
	const [properties, setProperties] = useState([])
	const [loading, setLoading] = useState(true)

	// console.log(loading, 'loading')
	// console.log(properties, 'properties')
	// console.log(propertyIdsFilter, 'propertyIdFilter')
	let city = window.location.pathname

	if (city == '/') {
		city = 'madrid'
	}

	const getPropertyType = async value => {
		try {
			const url = `/markers/${city}?type[]=${value}`
			const { data } = await axios(url)
			let ArrayDataString = []
			for (let i = 0; i < data.data.length; i++) {
				ArrayDataString.push(data.data[i].id)
				setPropertyIdsFilter(ArrayDataString)
			}
		} catch (error) {
			console.log(error)
		}
	}

	const getProperties = async value => {
		try {
			let urlIds = []
			if (propertyIdsFilter.length > 1) {
				for (let i = 0; i < 30; i++) {
					urlIds.push(`&ids[]=${propertyIdsFilter[i]}`)
				}
			}
			const urlIdsString = urlIds.toString().replaceAll(',', '').substring(1)
			const url = `/homecards_ids?${urlIdsString}`
			const { data } = await axios(url)
			const order = data.data.homecards

			order.sort(function (a, b) {
				if (a.pricePerMonth > b.pricePerMonth) {
					return 1
				}
			})

			if (value === 'ascending') {
				order.sort(function (a, b) {
					if (a.pricePerMonth > b.pricePerMonth) {
						return 1
					}
				})
			} else if (value === 'descending') {
				order.sort(function (a, b) {
					if (a.pricePerMonth < b.pricePerMonth) {
						return 1
					}
				})
			}
			setProperties(order)
		} catch (error) {
			console.log(error)
		}

		setLoading(false)
	}

	useEffect(() => {
		getProperties()
	}, [propertyIdsFilter])

	return (
		<PropertyTypeContext.Provider
			value={{
				properties,
				getPropertyType,
				getProperties,
				loading,
				propertyIdsFilter,
			}}
		>
			{children}
		</PropertyTypeContext.Provider>
	)
}

export { PropertyTypeProvider }

export default PropertyTypeContext
