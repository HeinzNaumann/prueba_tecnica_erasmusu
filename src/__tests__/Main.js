import { PropertyTypeProvider } from '../context/PropertyTypeProvider'
import { screen, render } from '@testing-library/react'
import Main from '../components/Main'

test('<Main probando los datos que vienen de la api', () => {
	render(
		<PropertyTypeProvider>
			<Main />
		</PropertyTypeProvider>
	)
	const properties = screen.findAllByTestId('properties')
	expect(properties).toMatchSnapshot()
})
