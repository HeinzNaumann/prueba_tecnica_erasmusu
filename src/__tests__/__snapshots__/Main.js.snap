// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main probando los datos que vienen de la api 1`] = `Promise {}`;
